hazardous/metrics/brier_score.py

```diff
@@ -0,0 +1,471 @@
+        "any" indicates that all events except the censoring 0 are
+        considered as a single event.
+        In single event settings, "any" and 1 are equivalent.
+
+    Returns
+    -------
+    times : np.ndarray of shape (n_times)
+        No-op, this is the same as the input.
+
+    brier_score : np.ndarray of shape (n_times)
+
+    References
+    ----------
+
+    [1] M. Kretowska, "Tree-based models for survival data with competing risks",
+        Computer Methods and Programs in Biomedicine 159 (2018) 185-198.
+    """
+    # XXX: make times an optional kwarg to be compatible with
+    # sksurv.metrics.brier_score?
+    # XXX: 'times' must match the times of y_pred,
+    # but we have no way to check that.
+    # In this sense, 'y_pred[:, t_idx]' is incorrect when 'times'
+    # is not the time used during the prediction.
+    computer = BrierScoreComputer(
+        y_train,
+        event_of_interest=event_of_interest,
+    )
+    return times, computer.brier_score_incidence(y_test, y_pred, times)
+
+
+def integrated_brier_score_incidence(
+    y_train,
+    y_test,
+    y_pred,
+    times,
+    event_of_interest="any",
+):
+    """Compute the Integrated Brier Score Incidence for the kth cause of failure.
+
+    .. math::
+
+        \\mathrm{IBS}_k(t) = \\frac{1}{t_{max} - t_{min}} \\int^{t_{max}}_{t_{min}}
+        \\mathrm{BS}_k(u) du
+
+    Parameters
+    ----------
+    y_train : record-array, dictionnary or dataframe of shape (n_samples, 2)
+        The target, consisting in the ``"event"`` and ``"duration"`` columns.
+        This is used to fit the IPCW estimator.
+
+    y_test : record-array, dictionnary or dataframe of shape (n_samples, 2)
+        The ground truth, consisting in the ``"event"`` and ``"duration"`` columns.
+
+    y_pred : array-like of shape (n_samples, n_times)
+        Incidence probability estimates predicted at ``times``.
+        In the binary event settings, this is 1 - survival_probability.
+
+    times : array-like of shape (n_times)
+        Times at which the survival probabilities ``y_pred`` has been estimated
+        and for which we compute the Brier Score.
+
+    event_of_interest : int or "any", default="any"
+        The event to consider in competitive events setting.
+        ``"any"`` indicates that all events except the censoring ``0`` are
+        considered as a single event.
+        In single event settings, ``"any"`` and ``1`` are equivalent.
+
+    Returns
+    -------
+    ibs : float
+
+    References
+    ----------
+
+    [1] M. Kretowska, "Tree-based models for survival data with competing risks",
+        Computer Methods and Programs in Biomedicine 159 (2018) 185-198.
+    """
+    times, brier_scores = brier_score_incidence(
+        y_train,
+        y_test,
+        y_pred,
+        times,
+        event_of_interest=event_of_interest,
+    )
+    return np.trapz(brier_scores, times) / (times[-1] - times[0])
+
+
+class BrierScoreSampler(BrierScoreComputer):
```

Comment:
I think this should not be part of the `hazardous.metrics` module, but instead could be moved into the source file of the estimator that uses it.

Same remark for the test.